File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minprice = prices[0];
        int maxProfit = 0;

        for(int i = 0; i < prices.size(); i++){
            if(prices[i] < minprice){
                minprice = prices[i];
            }else{
                maxProfit = max(maxProfit, prices[i]-minprice);
            }
        }

        return maxProfit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice = prices[0]
        maxSum = 0

        l = len(prices)

        for i in range(1,l):
            if minPrice > prices[i]:
                minPrice = prices[i]
            else: 
                maxSum = max(maxSum, prices[i] - minPrice)
        
        return maxSum